### PR TITLE
cd: fix cleanup logic install.sh

### LINKED ISSUE #53: install.sh performs complete cleanup when interrupting autolauched streamlit app
As per screenshot, Ctrl+C leads to the full cleanup of project's directory:
image

### PR DESCRIPTION
**Pull request by DanilKhardi**
## Description

Proposed changes fix situation when install.sh performs complete cleanup when interrupting autolauched streamlit app

## Related Issue

Fixes #53

## Proposed Changes

- Disables complete removal of project's directory.
- Adds `SIGINT` as trapped signal for `cleanup` function.

## Checklist

- [x] I have tested these changes locally.
- [x] I have updated the relevant documentation.
- [x] I have added necessary tests if needed.
- [x] I have followed the project's coding conventions.
- [x] I have checked for any breaking changes.

## Additional Notes

No additional notes.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -21,12 +21,11 @@ command_exists() {
 cleanup() {
     echo "Cleaning up..."
     deactivate >/dev/null 2>&1
-    rm -rf "$HOME/$dir_name"
```
Comment (khit-mle) [inline install.sh:24]: I agree that we should stop forceful removal of `$HOME/$dir_name` on cleanup.

**Review by khit-mle — COMMENTED** [review]
I also propose we modify this script so that it could also accept as arguments:
1) the need to add LLM summarizer functionality;
2) the need to run streamlit app after main script tasks are executed.

**Comment by alexa313** [comment]
I think we should switch to parsing named arguments from script execution:
```bash
# Parse command line arguments
for arg in "$@"; do
    case $arg in
        --dir_name=*)
            dir_name="${arg#*=}"
            ;;
        --git_branch=*)
            branch="${arg#*=}"
            ;;
        --llm_summarizer=*)
            llm_summarizer="${arg#*=}"
            ;;
        --run_app=*)
            run_app="${arg#*=}"
            ;;
        --llm_api_key=*)
            LLM_API_KEY="${arg#*=}"
            ;;
        --llm_url=*)
            LLM_URL="${arg#*=}"
            ;;
    esac
done
```

**Review by khit-mle — APPROVED** [review]
## Code Review of [PR-52](https://github.com/mlteamurfu2325/practicum-s1/pull/52) by @khit-mle 

### Overview
- [x] The code follows the project's coding conventions and style guide.
- [x] The code is well-structured, readable, and maintainable.
- [x] The code is properly documented with clear comments and docstrings.
- [x] The code is efficient and optimized for performance, if applicable.
- [x] The code handles edge cases and potential errors gracefully.
- [x] The code is free of any security vulnerabilities or potential risks.

### Functionality
- [x] The code implements the required functionality as described in the PR description.
- [x] The code handles all expected inputs and outputs correctly.
- [x] The code integrates well with the existing codebase and does not introduce any conflicts.
- [x] The code does not introduce any new bugs or regressions.

### Testing
- [x] The code passes all existing tests and any newly added tests.
- [x] The tests cover a sufficient range of scenarios and edge cases (needed if new tests are added).
- [x] The tests are well-structured, readable, and maintainable (needed if new tests are added).

### Documentation
- [x] The code includes clear and concise comments explaining complex logic or algorithms.
- [x] The code includes docstrings for functions, classes, and modules, following the project's documentation standards.
- [x] The PR description provides a clear explanation of the changes made and their purpose.
- [x] Any necessary updates to the project's documentation (README, API docs, etc.) are included.

### Suggestions and Improvements
- [ ] Are there any areas where the code could be further optimized or improved?
- [ ] Are there any best practices or design patterns that could be applied to enhance the code?
- [ ] Are there any potential edge cases or error scenarios that have not been considered?
- [ ] Are there any additional tests that could be added to improve code coverage?

### Other Comments
@DanilKhardi  Thank you for your PR#52.

## Overall Recommendation:
- [x] Approve
- [ ] Request Changes
- [ ] Comment Only

**Review by alexa313 — APPROVED** [review]
## Code Review of [PR-52](https://github.com/mlteamurfu2325/practicum-s1/pull/52) by @alexa313 

### Overview
- [x] The code follows the project's coding conventions and style guide.
- [x] The code is well-structured, readable, and maintainable.
- [x] The code is properly documented with clear comments and docstrings.
- [x] The code is efficient and optimized for performance, if applicable.
- [x] The code handles edge cases and potential errors gracefully.
- [x] The code is free of any security vulnerabilities or potential risks.

### Functionality
- [x] The code implements the required functionality as described in the PR description.
- [x] The code handles all expected inputs and outputs correctly.
- [x] The code integrates well with the existing codebase and does not introduce any conflicts.
- [x] The code does not introduce any new bugs or regressions.

### Testing
- [x] The code passes all existing tests and any newly added tests.
- [x] The tests cover a sufficient range of scenarios and edge cases (needed if new tests are added).
- [x] The tests are well-structured, readable, and maintainable (needed if new tests are added).

### Documentation
- [x] The code includes clear and concise comments explaining complex logic or algorithms.
- [x] The code includes docstrings for functions, classes, and modules, following the project's documentation standards.
- [x] The PR description provides a clear explanation of the changes made and their purpose.
- [x] Any necessary updates to the project's documentation (README, API docs, etc.) are included.

### Suggestions and Improvements
- [ ] Are there any areas where the code could be further optimized or improved?
- [ ] Are there any best practices or design patterns that could be applied to enhance the code?
- [ ] Are there any potential edge cases or error scenarios that have not been considered?
- [ ] Are there any additional tests that could be added to improve code coverage?

### Other Comments
@DanilKhardi Everything works as intended.

## Overall Recommendation:
- [x] Approve
- [ ] Request Changes
- [ ] Comment Only
